components/ai_chat/content/browser/ai_chat_tab_helper.cc

```diff
@@ -48,13 +73,29 @@ AIChatTabHelper::AIChatTabHelper(
           web_contents->GetBrowserContext()
               ->GetDefaultStoragePartition()
               ->GetURLLoaderFactoryForBrowserProcess(),
-          channel_name) {
+          channel_name),
+      pdf_load_observer_(
+          std::make_unique<PDFA11yInfoLoadObserver>(nullptr, this)) {
   favicon::ContentFaviconDriver::FromWebContents(web_contents)
       ->AddObserver(this);
 }
 
 AIChatTabHelper::~AIChatTabHelper() = default;
 
+void AIChatTabHelper::SetOnPDFA11yInfoLoadedCallbackForTesting(
+    base::OnceClosure cb) {
+  on_pdf_a11y_info_loaded_cb_ = std::move(cb);
+}
+
+void AIChatTabHelper::OnPDFA11yInfoLoaded() {
+  DVLOG(3) << "PDF Loaded";
+  MaybeGeneratePageText();
```

Comment:
Just to let you know that as of https://github.com/brave/brave-core/pull/21408, we don't need to call `MaybeGeneratePageText` and possibly you won't need a "Loaded" event anymore. The text is now retrieved when needed, on every message send.

Reply:
No, we need that pdf loaded event, otherwise pdf extraction will fetch nothing when users click summarize this page. This can be easily reproduced with large pdf file.

Address it in https://github.com/brave/brave-core/pull/21570/commits/2b035c66a4a4d5fccd4b188ae353273d3c4f2093
When pdf is not ready, users won't see suggested questions until it is.